### PR TITLE
C374151 Verify Bulk edit state when navigating to another app and bac…

### PR DESCRIPTION
**Pull request by KarapetyanSofia**
[C374151](https://foliotest.testrail.io/index.php?/cases/view/374151)

[Allure Report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/5594/allure)

image

### CODE REVIEW

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        BulkEditSearchPane.selectRecordIdentifier('Holdings UUIDs');
+        BulkEditSearchPane.verifyDragNDropHoldingsUUIDsArea();
+
+        BulkEditSearchPane.uploadFile(validHoldingUUIDsFileName);
+        BulkEditSearchPane.checkForUploading(validHoldingUUIDsFileName);
+        BulkEditSearchPane.waitFileUploading();
+
+        TopMenuNavigation.navigateToApp('Inventory');
+        InventorySearchAndFilter.switchToHoldings();
+        InventorySearchAndFilter.searchHoldingsByUUID(item.holdingUUID);
```
Comment (manowarqqq) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:70]: No need new method
        InventorySearchAndFilter.searchByParameter('Holdings UUID', item.holdingUUID);

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        TopMenuNavigation.navigateToApp('Bulk edit');
```
Comment (manowarqqq) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:72]: No expected result for Step 4

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        cy.reload();
```
Comment (manowarqqq) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:73]: No expected result for Step 5

File: cypress/support/fragments/inventory/inventorySearchAndFilter.js [hunk 4]
```diff
@@ -99,6 +99,15 @@ const searchHoldingsByHRID = (hrid) => {
   InventoryInstances.waitLoading();
 };
 
+const searchHoldingsByUUID = (uuid) => {
```
Comment (manowarqqq) [inline cypress/support/fragments/inventory/inventorySearchAndFilter.js:102]: already exists InventorySearchAndFilter.searchByParameter

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js (hunk 2 above)
Comment (KarapetyanSofia) [reply cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:74]: Done!

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+const item = {
+  instanceName: `testBulkEdit_${getRandomPostfix()}`,
+  itemBarcode: getRandomPostfix(),
+  instanceId: '',
+  holdingUUID: '',
+  holdingHRID: '',
+};
+
+describe('Bulk Edits', () => {
+  describe('Bulk Edit - Holdings', () => {
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:22]: make sure describe blocks' texts match with other tests in the same folder

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import TopMenuNavigation from '../../../support/fragments/topMenuNavigation';
+
+const testData = {};
+const validHoldingUUIDsFileName = `SearchHoldingUUIDs_${getRandomPostfix()}.csv`;
+const item = {
+  instanceName: `testBulkEdit_${getRandomPostfix()}`,
+  itemBarcode: getRandomPostfix(),
+  instanceId: '',
+  holdingUUID: '',
+  holdingHRID: '',
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:18]: do you need these initialized here? instanceId: '',
  holdingUUID: '',
  holdingHRID: '',

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        BulkEditSearchPane.selectRecordIdentifier('Holdings UUIDs');
+        BulkEditSearchPane.verifyDragNDropHoldingsUUIDsArea();
+
+        BulkEditSearchPane.uploadFile(validHoldingUUIDsFileName);
+        BulkEditSearchPane.checkForUploading(validHoldingUUIDsFileName);
+        BulkEditSearchPane.waitFileUploading();
+
+        TopMenuNavigation.navigateToApp('Inventory');
+        InventorySearchAndFilter.switchToHoldings();
+        InventorySearchAndFilter.searchByParameter('Holdings UUID', item.holdingUUID);
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:72]: are you verifying that the record is found?

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        cy.wait(10000); // need to navigate and verify matched result
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:73]: does it not work without waiting for 10 seconds?

Comment (KarapetyanSofia) [reply cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:73]: No, I`ve tried, but it doesn`t

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js (hunk 6 above)
Comment (KarapetyanSofia) [reply cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:18]: Yes, it needs me!

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js (hunk 8 above)
Comment (manowarqqq) [reply cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:73]: Check, that instance exists
   InventorySearchAndFilter.verifyInstanceDisplayed(item.instanceName);
and remove wait()

**Review by manowarqqq — COMMENTED** [review]
.

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js (hunk 8 above)
Comment (KarapetyanSofia) [reply cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-state-during-navigation.cy.js:76]: @manowarqqq Thanks so much!

**Review by manowarqqq — APPROVED** [review]

**Review by nayimovag — APPROVED** [review]

**Pull request merged**